### PR TITLE
Add the file close operation before function return to advoid resource leaking

### PR DESCRIPTION
**Pull request by MabinGo**
In some functions,they do the open file operations but do not be close before functions return,so close file operations  should be added to advoid resource leaking.

### CODE REVIEW

File: daemon/graphdriver/devmapper/mount.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ func ProbeFsType(device string) (string, error) {
 	buffer := make([]byte, maxLen)
 	l, err := file.Read(buffer)
 	if err != nil {
+		file.Close()
```
Comment (ahmetalpbalkan) [inline daemon/graphdriver/devmapper/mount.go]: shouldn't this be a `defer file.Close()` just after the previous err check?

File: vendor/src/github.com/docker/libcontainer/notify_linux.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ func NotifyOnOOM(s *State) (<-chan struct{}, error) {
 	}
 	fd, _, syserr := syscall.RawSyscall(syscall.SYS_EVENTFD2, 0, syscall.FD_CLOEXEC, 0)
 	if syserr != 0 {
+		oomControl.Close()
```
Comment (ahmetalpbalkan) [inline vendor/src/github.com/docker/libcontainer/notify_linux.go]: must use `defer`, just after `os.Open`'s error check

File: daemon/graphdriver/devmapper/mount.go (hunk 1 above)
Comment (MabinGo) [inline daemon/graphdriver/devmapper/mount.go]: use defer file.Close() after the previous err check is also ok, however,shall we use current modify to keep the coding style accordnate to the original?

Comment (ahmetalpbalkan) [inline daemon/graphdriver/devmapper/mount.go]: @MabinGo usage of `defer` is idiomatic in go. it provides certain guarantees: http://blog.golang.org/defer-panic-and-recover it's no big deal to add defer in these cases. I'm not sure what we gain by preserving original style. It was missing file.Close() and you can add them using `defer`.

Comment (MabinGo) [inline daemon/graphdriver/devmapper/mount.go]: ok,thanks:) 
do i create another PR to update the code?

Comment (ahmetalpbalkan) [inline daemon/graphdriver/devmapper/mount.go]: nope you just make changes on your local branch, `git add` those, then `git commit --amend --no-edit` and `git push -f your-remote checkfileoper1` and this PR will be updated automatically.

Comment (MabinGo) [inline daemon/graphdriver/devmapper/mount.go]: thanks:) 
the code is updated and commited.

**Comment by ahmetalpbalkan** [comment]
LGTM (not a maintainer).

**Comment by phemmer** [comment]
Just a side comment, while this may be the proper way of handling, it shouldn't have any effect on leaking resources. Go adds a finalizer on File objects so that they get closed if they are garbage collected: http://golang.org/src/os/file_unix.go?s=1063:1106#L47

**Comment by estesp** [comment]
Thanks for your submission, however, libcontainer is vendored into the docker source tree via a controlled process.  PRs against `docker/docker` cannot contain modifications to anything in the `vendor` subtree.  If you wish to make changes to libcontainer, you need to open a PR in `docker/libcontainer`.

Thanks!

**Comment by MabinGo** [comment]
@phemmer Ok,the user should open and close files explicitly.

@estesp Thanks to remind me, i will keep the modify of daemon/graphdriver/devmapper/mount.go only,and rollback modify of vendor/src/github.com/docker/libcontainer/notify_linux.go

**Comment by thaJeztah** [comment]
@MabinGo did you create a pull-request https://github.com/docker/libcontainer as well?

**Comment by MabinGo** [comment]
@thaJeztah I have not created PR in https://github.com/docker/libcontainer yet, i thought  the PR against docker/docker would be reviewed first:)

**Comment by thaJeztah** [comment]
@MabinGo alright, thanks!

**Comment by MabinGo** [comment]
ping @estesp I have updated the PR #11054 to keep the fix against docker/docker only,could you please review the PR again? Thanks:)

**Comment by thaJeztah** [comment]
@MabinGo I think you already "mentioned" him after you updated the PR, so not necessary to 'ping' again :smile:

**Comment by MabinGo** [comment]
@thaJeztah  OK,thanks: )

**Comment by estesp** [comment]
LGTM

**Comment by duglin** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM

@duglin @estesp you can merge next time ;) no sense in waiting for so small a change
